source/extensions/filters/network/mongo_proxy/proxy.cc

```diff
@@ -81,11 +81,9 @@ void ProxyFilter::setDynamicMetadata(std::string operation, std::string resource
             .dynamicMetadata()
             .mutable_filter_metadata())[NetworkFilterNames::get().MongoProxy]);
   auto& fields = *metadata.mutable_fields();
-  auto& list = *fields[DynamicMetadataKeysSingleton::get().MessagesField].mutable_list_value();
-  auto& message = *list.add_values()->mutable_struct_value()->mutable_fields();
-
-  message[DynamicMetadataKeysSingleton::get().OperationField].set_string_value(operation);
-  message[DynamicMetadataKeysSingleton::get().ResourceField].set_string_value(resource);
+  // TODO(rshriram): reverse the resource string (table.db)
```

Comment:
Super nit, add period at the end of the sentence (and also the other TODOs/comments).